src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -244,6 +244,27 @@ protected void setBaseCaseSensitivityValues(List<SensitivityFactorGroup<DcVariab
         }
     }
 
+    private void createBranchFunctionReferenceValue(LfSensitivityFactor<DcVariableType, DcEquationType> factor, DenseMatrix contingenciesStates,
+                                                    Collection<ComputedContingencyElement> contingencyElements,
+                                                    PropagatedContingency contingency, SensitivityValueWriter valueWriter) {
+        double flowValue;
+        EquationTerm<DcVariableType, DcEquationType> p1 = factor.getFunctionEquationTerm();
+        String functionBranchId = factor.getFunctionElement().getId();
+        flowValue = factor.getFunctionReference();
+        for (ComputedContingencyElement contingencyElement : contingencyElements) {
+            double contingencySensitivity = p1.calculateSensi(contingenciesStates, contingencyElement.getContingencyIndex());
+            flowValue += contingencyElement.getAlphaForFunctionReference() * contingencySensitivity;
+            if (contingencyElement.getElement().getId().equals(functionBranchId)) {
+                // the monitored branch is in contingency, its post-contingency flow equals to zero.
+                flowValue = 0d;
+                break;
+            }
+        }
+
+        valueWriter.write(factor.getContext(), contingency != null ? contingency.getContingency().getId() : null, contingency != null ? contingency.getIndex() : -1,
+                0d, unscaleFunction(factor, flowValue));
```

Comment:
Here we choose `0` for sensitivity value, it is what you want? Or is it better to choose `Double.NaN`?